intf/cnsr_intf.lua

```diff
@@ -10,69 +11,72 @@ FRAME_INTERVAL = 30000
 SKIP_SAFETY = 10000
 MINIMUM_DISPLAY_TIME = 2000000
 GET_CONFIG_INTERVAL = 500
+-- end constants
 
 
 DESCRIPTIONS = { [1] = "violence",
 			   [2] = "verbal abuse",
 			   [3] =  "nudity",
 			   [4] =  "alcohol and drug consumption"}
 
-
-tags = {}
+-- globals
 tag_index = 1
 current_time = 0
 prev_time = 0
-loop_counter = 0
 reverse = false
 done = false
-
-mute_params = {}
-mute_params.start_time = 0
-mute_params.end_time = 0
-mute_params.activated = false
-mute_params.prev_volume = vlc.volume.get()
-mute_params.action_word = "muted"
-
-function mute()
-	mute_params.prev_volume = vlc.volume.get()
+-- end globals
+
+-- mute
+MuteParams = {
+	start_time = 0,
+	end_time = 0,
+	activated = false,
+	prev_volume = vlc.volume.get(),
+	action_word = "muted"
+}
+
+function MuteParams.mute()
+	MuteParams.prev_volume = vlc.volume.get()
 	vlc.volume.set(0)
 end
+-- end mute
 
-mute_params.activate = mute
-
-hide_params = {}
-hide_params.start_time = 0
-hide_params.end_time = 0
-hide_params.activated = false
-hide_params.action_word = "hidden"
+-- hide
+HideParams = {
+	start_time = 0,
+	end_time = 0,
+	activated = false,
+	action_word = "hidden"
+}
 
-function hide()
+function HideParams.activate()
 	hide_filter = vlc.object.vout()
 	vlc.var.create(hide_filter, "contrast", 0)
 	vlc.var.set(hide_filter, "video-filter", "adjust")
 end
+-- end hide
 
-hide_params.activate = hide
-
-
-
-function Looper()
-	next_loop_time = vlc.misc.mdate()
+-- Main app entry point
+function looper()
+	local next_loop_time
+	local loop_counter = 0
+	local tags;
 	while true do
 		if vlc.volume.get() == -256 then break end  -- inspired by syncplay.lua; kills vlc.exe process in Task Manager
 		if loop_counter == 0 then
-			Get_config() -- We don't want to call it more then we have to.
-			Log("got config")
+			get_config() -- We don't want to call it more then we have to.
+			log("got config")
 		end
 		loop_counter = (loop_counter + 1) % GET_CONFIG_INTERVAL
 		if vlc.playlist.status()~="stopped" and config.CNSR and config.CNSR.tags and #config.CNSR.tags ~= 0 then
 			tags = config.CNSR.tags
 
-			input = vlc.object.input()
+			local input = vlc.object.input()
 			current_time = vlc.var.get(input,"time")
 			check_disable_actions()
 			reverse = prev_time > current_time
-			tag = get_current_tag()
+			local tag = get_current_tag()
```

Comment:
should receive "tags" since they are no longer global

Reply:
done